src/interface/mod.rs

```diff
@@ -1,7 +1,20 @@
 // Author: Nicholas Renner
-//
-// Module definitions for the SafePOSIX Rust interface
-// this interface limits kernel access from Rust to the popular paths as defined in Lock-in-Pop
+//! Module definitions for the RustPOSIX interface
+//! 
+//! ## Interface Module
+//!
+//! Secure interface module that enforces containment of kernel calls to "popular paths" to enhance security. It restricts access to libraries only through specified paths in order to limit kernel calls to these popular paths.
+//! The RustPOSIX interface exposes RustPOSIX functionalities to libraries through five specific files:
```

Comment:
Do we hide this or expose it to people using it?  If we hide it, we don't need to explain this structure in rustdoc.  It could be a normal comment.

In other words, if I need to know this to write a program that uses this module, you should tell me.  If not, then you can hide this detail.

Reply:
understood. will refactor